src/common/scan-info/src/expr_rewriter.rs

```diff
@@ -173,7 +172,8 @@ pub fn rewrite_predicate_for_partitioning(
                 ref left,
                 ref right,
             } => {
-                if let Expr::Column(col_name) = left.as_ref()
+                if let Expr::Column(Column::Resolved(ResolvedColumn::Basic(col_name))) =
```

Comment:
I wonder if we should introduce some syntactic sugar on expr that lets us do `if expr.is_column_name(col_name)`. Then we don't have to do this nested matching for every case.

Reply:
It looks like this file was the majority of the boilerplate so I made a helper closure to fix that